tests/api/test_links.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+from fastapi import APIRouter, FastAPI
+from starlette.requests import Request
+from starlette.testclient import TestClient
+
+from stac_fastapi.pgstac.models import links as app_links
+
+
+@pytest.mark.parametrize("root_path", ["", "/api/v1"])
+@pytest.mark.parametrize("prefix", ["", "/stac"])
+def tests_app_links(prefix, root_path):
+    endpoint_prefix = root_path + prefix
+    url_prefix = "http://stac.io" + endpoint_prefix
```

Comment:
see #201 , but this should also work:

```suggestion
    endpoint_prefix = prefix
    url_prefix = "http://stac.io" + root_path + endpoint_prefix
```

we want to model situation when `root_path` is dropped by the reverse proxy, but needs to be present in the output links as this is the only way to reach the API.

Reply:
🤔 I'm not sure to get this to be honest

In all cases, the client should be called `{root-path}/{prefix}/endpoint` as shown in https://github.com/encode/starlette/blob/a7d0b14c7378aa2e95b0b13583fe0dadace363be/tests/test_routing.py#L573-L575

Reply:
> 🤔 I'm not sure to get this to be honest
> 
> In all cases, the client should be called `{root-path}/{prefix}/endpoint` as shown in https://github.com/encode/starlette/blob/a7d0b14c7378aa2e95b0b13583fe0dadace363be/tests/test_routing.py#L573-L575
> 
> 

I don't understand how you come to this conclusion from that one test alone. I'm just trying to replicate the scenario I see in the debugger of a real system. Omitting the root path in the call to the test client still routes to the endpoint in question, but this time with the request object behaving the same way as in the real system behind a reverse proxy that removes routing prefix.